api/app/tests/apps/api/v1/test_filters.py

```diff
@@ -1010,15 +1026,28 @@ def test_retrieve_all_parents_with_changes_in_children(self):
 
     def test_retrieve_all_parents_with_no_changes_in_children(self):
         now = timezone.now()
-        with freeze_time(now - timedelta(hours=1)):
-            parent_signal = SignalFactory()
 
-        with freeze_time(now):
-            SignalFactory(parent=parent_signal)
+        parent_signal = SignalFactory.create()
+        for hour in range(5):
+            # Create 4 child signals 1 hour apart
+            with freeze_time(now + timedelta(hours=hour)):
+                SignalFactory.create(parent=parent_signal)
 
-        with freeze_time(now + timedelta(hours=1)):
+        with freeze_time(now + timedelta(hours=6)):
+            # This way the parent signal is changed last
             parent_signal.save()
 
+        # This parent should not show up in the filter
+        parent_signal_not_to_be_filtered = SignalFactory.create()
+        for hour in range(5):
+            # Create 4 child signals 1 hour apart
+            with freeze_time(now + timedelta(hours=hour)):
+                SignalFactory.create(parent=parent_signal_not_to_be_filtered)
+
+        with freeze_time(now + timedelta(hours=4)):
```

Comment:
Is using `hours=4` not ambiguous here? The loop variable `hour` in the previous loop where children are created also reaches `4`. For this test I think using `3.5` reads better. Also I think we have to carefully consider the behavior when `parent_signal.updated_at == child_signal.updated_at`.

Reply:
Should have been 3, changed it